Content.Server/Explosion/EntitySystems/TriggerSystem.TimedCollide.cs

```diff
@@ -31,27 +29,26 @@ private void OnTimerEndCollide(EntityUid uid, TriggerOnTimedCollideComponent com
 
         if (component.Colliding.Count == 0)
         {
-            Active.Remove(uid);
+            RemComp<ActiveTriggerOnTimedCollideComponent>(uid);
         }
     }
 
     private void OnComponentRemove(EntityUid uid, TriggerOnTimedCollideComponent component, ComponentRemove args)
     {
-        Active.Remove(uid);
+        RemComp<ActiveTriggerOnTimedCollideComponent>(uid);
     }
 
     private void UpdateTimedCollide(float frameTime)
     {
-        foreach (var trigger in Active)
+        foreach (var (activeTrigger, triggerOnTimedCollide) in EntityQuery<ActiveTriggerOnTimedCollideComponent, TriggerOnTimedCollideComponent>())
         {
-            if (!TryComp(trigger, out TriggerOnTimedCollideComponent? component))
-                continue;
-            foreach (var (collidingEntity, collidingTimer) in component.Colliding)
+            foreach (var (collidingEntity, collidingTimer) in triggerOnTimedCollide.Colliding)
             {
-                component.Colliding[collidingEntity] += frameTime;
-                if (collidingTimer > component.Threshold)
+                triggerOnTimedCollide.Colliding[collidingEntity] += frameTime;
+                if (collidingTimer > triggerOnTimedCollide.Threshold)
                 {
-                    RaiseLocalEvent(trigger, new TriggerEvent(trigger, collidingEntity));
+                    RaiseLocalEvent(activeTrigger.Owner, new TriggerEvent(activeTrigger.Owner, collidingEntity));
+                    triggerOnTimedCollide.Colliding[collidingEntity] = 0;
```

Comment:
```suggestion
                    triggerOnTimedCollide.Colliding[collidingEntity] -= triggerOnTimedCollide.Threshold;
```
This prevents time drift. Especially with lower tick rates.